src/main/java/com/google/sps/servlets/MailServlet.java

```diff
@@ -0,0 +1,150 @@
+
+// [START multipart_includes]
+import java.io.InputStream;
+import java.io.ByteArrayInputStream;
+import java.io.UnsupportedEncodingException;
+import javax.activation.DataHandler;
+import javax.mail.Multipart;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMultipart;
+// [END multipart_includes]
+
+import com.google.sps.data.Postcard;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.util.Map;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/mail")
+public class MailServlet extends HttpServlet {
+
+  Map<String, Postcard> emails;
+
+  public void init() {
+     emails = new HashMap<String, Postcard>();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+
+    UserService userService = UserServiceFactory.getUserService();
+    String from = userService.getCurrentUser().getEmail();
+
+    Postcard card = emails.get(from);
+    String to = card.to;
+    String text = card.text;
+    String link = card.link;
+
+    sendMultipartMail(from, to, link, text);
+
+    resp.setContentType("text/html;");
+    resp.getWriter().println("Your postcard has been sent!");
+  }
+
+  private void sendMultipartMail(String from, String to, String link, String text) {
+    Properties props = new Properties();
+    Session session = Session.getDefaultInstance(props, null);
+
+    String msgBody = "...";
+
+    try {
+      Message msg = new MimeMessage(session);
+      msg.setFrom(new InternetAddress(from, "Example.com Admin"));
+      msg.addRecipient(Message.RecipientType.TO,
+                       new InternetAddress(to, "Mr. User"));
+      msg.setSubject("You've received a postcard!");
+      msg.setText(msgBody);
+
+      String htmlBody = "<div style=\"display:flex;flex-direction:row;justify-content:center;\">" +
+                    "<img src = " + "\"" + link + "\"" + "style=\"width:400px\">" +
+                    "<div style=\"width:200px; border:thin solid black;\">" +
+                      "<text style=\"display:flex;flex-wrap:wrap;justify-content:center;padding:30px;\">" + text + "</text>" +
+                    "</div>" +
+                  "</div>";
+
+      // [START multipart_example]
+      //String htmlBody = "<img src = " + "\"" + link + "\"" + " style=\"height:200px\"" + ">";
+      //String htmlBody = "<img id=\"myImage\" src = \"https://www.w3schools.com/js/pic_bulboff.gif\" style=\"width:500px\">";          // ...
+      byte[] attachmentData = null;  // ...
+      Multipart mp = new MimeMultipart();
+
+      MimeBodyPart htmlPart = new MimeBodyPart();
+      htmlPart.setContent(htmlBody, "text/html");
+      mp.addBodyPart(htmlPart);
+
+      /*MimeBodyPart attachment = new MimeBodyPart();
```

Comment:
I think you can remove the commented code as well. It is not necessary to keep code that is not used.